app/controllers/products_controller.rb

```diff
@@ -1,37 +1,34 @@
 class ProductsController < ApplicationController
-  before_action :require_login, except: [:index, :show, :merchant_products]
+  before_action :require_login, only: [:new, :create, :edit, :update, :destroy, :retire]
+  before_action :find_product,  only: [:show, :edit, :update, :destroy, :retire]
 
   def index
     @products = Product.active_product
```

Comment:
It seems like this method is returning a list so a plural method name might make more sense